fixups/EnvVarFixup/EnvVarFixup.cpp

```diff
@@ -0,0 +1,376 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Tim Mangan. All rights reserved
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include <regex>
+#include <psf_framework.h>
+#include "FunctionImplementations.h"
+#include "EnvVar_spec.h"
+
+
+extern std::vector<env_var_spec> g_envvar_envVarSpecs;
+
+DWORD g_EnvVarInterceptInstance = 0;
+
+
+//ExpandEnvironmentStrings
+//  [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
+//  public static extern int ExpandEnvironmentStrings([MarshalAs(UnmanagedType.LPTStr)] String source, [Out] StringBuilder destination, int size);
+
+
+//GetEnvironmentVariable
+//      DllImport("kernel32.dll")]
+//  static extern uint GetEnvironmentVariable(string lpName,
+//      [Out] StringBuilder lpBuffer, uint nSize);
+//
+auto GetEnvironmentVariableImpl = psf::detoured_string_function(&::GetEnvironmentVariableA, &::GetEnvironmentVariableW);
+template <typename CharC, typename CharT>
+DWORD __stdcall GetEnvironmentVariableFixup(_In_ const CharC* lpName, _Inout_ CharT* lpValue, _In_ DWORD lenBuf)
+{
+#if _DEBUG
+    DWORD GetEnvVarInstance = ++g_EnvVarInterceptInstance;
+    LogString(GetEnvVarInstance,"GetEnvironmentVariableFixup called for", lpName);
+#endif
+    auto guard = g_reentrancyGuard.enter();
+    DWORD result;
+
+    if (guard)
+    {
+#if _DEBUG
+        Log("[%d] GetEnvironmentVariableFixup unguarded.", GetEnvVarInstance);
+#endif
+        std::wstring eName;
+        if constexpr (psf::is_ansi<CharT>)
+        {
+            eName = widen(lpName);
+        }
+        else
+        {
+            eName =lpName;
+        }

```

Comment:
Add a line break after the closing bracket.

Reply:
Done